src/wallet/signer.ts

```diff
@@ -0,0 +1,192 @@
+ * that then gets signed and returned.
+ *
+ * @param transactions - An array of Transactions (in object or blob form) to combine and sign.
+ * @returns A single signed Transaction which has all Signers from transactions within it.
+ * @throws ValidationError if:
+ * - There were no transactions given to sign
+ * - The SigningPubKey field is not the empty string in any given transaction
+ * - Any transaction is missing a Signers field.
+ */
+function multisign(transactions: Array<Transaction | string>): Transaction {
+  if (transactions.length === 0) {
+    throw new ValidationError("There were 0 transactions to multisign");
+  }
+
+  transactions.forEach((txOrBlob) => {
+    const tx: Transaction = getDecodedTransaction(txOrBlob);
+
+    // This will throw a more clear error for JS users if any of the supplied transactions has incorrect formatting
+    // TODO: Replace this with verify() (The general validation function for all Transactions)
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- verify does not accept Transaction type
+    verifyBaseTransaction(tx as unknown as Record<string, unknown>);
+
+    if (tx.SigningPubKey !== "") {
+      throw new ValidationError(
+        "SigningPubKey must be an empty string for all transactions when multisigning.s"
+      );
+    }
+
+    if (tx.Signers == null || tx.Signers.length === 0) {
+      throw new ValidationError(
+        "For multisigning all transactions must include a Signers field containing an array of signatures. You may have forgotten to pass the 'multisign' parameter when signing."
+      );
+    }
+  });
+
+  const decodedTransactions: Transaction[] = transactions.map(
+    (txOrBlob: string | Transaction) => {
+      return getDecodedTransaction(txOrBlob);
+    }
+  );
+
+  validateTransactionEquivalence(decodedTransactions);
+
+  return getTransactionWithAllSigners(decodedTransactions);
+}
+
+/**
+ * Creates a signature that can be used to redeem a specific amount of XRP from a payment channel.
+ *
+ * @param wallet - The account that will sign for this payment channel.
+ * @param channelId - An id for the payment channel to redeem XRP from.
+ * @param amount - The amount in drops to redeem.
+ * @returns A signature that can be used to redeem a specific amount of XRP from a payment channel.
+ */
+function authorizeChannel(
+  wallet: Wallet,
+  channelId: string,
+  amount: string
+): string {
+  const signingData = encodeForSigningClaim({
+    channel: channelId,
+    amount,
+  });
+
+  return signWithKeypair(signingData, wallet.privateKey);
+}
+
+/**
+ * Verifies that the given transaction has a valid signature based on public-key encryption.
+ *
+ * @param tx - A transaction to verify the signature of. (Can be in object or encoded string format).
+ * @returns Returns true if tx has a valid signature, and returns false otherwise.
+ */
+function verify(tx: Transaction | string): boolean {
```

Comment:
This should be `verifySignature` or `verifySignedTransaction` or something slightly more descriptive.

Reply:
I think `verifySignature` makes the most sense, because that's the only check it does and I don't want to confuse it with the more general verify(Tx)